### PR TITLE
Update to Play 2.5

### PR DESCRIPTION
**Pull request by AlanQuatermain**
- Deprecates all APIs using play GlobalSettings type; hard delete & replacement will come in v2.6.
- Provide new constructors without GlobalSettings.
- Removed all uses of play's FakeApplication type; we now use default GuiceBuilder in all cases.

### CODE REVIEW

**Comment by AlanQuatermain** [comment]
NB: This bumps the plugin's version to 2.5 (see build.sbt)

File: build.sbt [hunk 1]
```diff
@@ -10,11 +10,10 @@ lazy val NGHelpers = (project in file("helpers"))
   .settings(
     name := "play-testng-helpers",
     scalaVersion := "2.11.11",
-    crossScalaVersions := Seq("2.11.11", "2.10.6"),
```
Comment (FranklinYinanDing) [inline build.sbt]: Please change this to `crossScalaVersions := Seq("2.11.11", "2.12.4"),`, since we need to support both Scala 2.11 and 2.12

Comment (FranklinYinanDing) [inline build.sbt]: Please change this to crossScalaVersions := Seq("2.11.11", "2.12.4"),, since we need to support both Scala 2.11 and 2.12.

File: helpers/src/main/java/com/linkedin/plugin/NGTestsBase.java [hunk 2]
```diff
@@ -101,14 +101,18 @@ public T getTestServerAnnotation() {
       return conf;
     }
 
+    /**
+     * @deprecated Use dependency injection with Play 2.4 instead of plugins.
+     */
+    @Deprecated
     protected List<String> getPlugins(){
```
Comment (FranklinYinanDing) [inline helpers/src/main/java/com/linkedin/plugin/NGTestsBase.java]: This should be removed. Play 2.5 no longer supports Plugin.

File: plugin/src/main/scala/NGPlugin.scala [hunk 3]
```diff
@@ -20,17 +20,43 @@ import de.johoop.testngplugin.TestNGPlugin._
...
      testNGParameters ++= Seq("-listener", "com.linkedin.plugin.FailSkippedTestsListener"),
-       libraryDependencies <++= (testNGVersion)(v => Seq(
+       libraryDependencies ++= testNGVersion(v => Seq(
          "org.testng" % "testng" % v % "test->default",
          // If changing this, be sure to change in Build.scala also.
-         "de.johoop" %% "sbt-testng-interface" % "3.0.2" % "test"))
+         "de.johoop" %% "sbt-testng-interface" % "3.0.2" % "test")).value
   )
+
+  def ngProjectSettings: Seq[Setting[_]] = super.projectSettings ++ Seq(
```
Comment (FranklinYinanDing) [inline plugin/src/main/scala/NGPlugin.scala]: Please see [here](http://www.scala-sbt.org/0.13/docs/Plugins.html#projectSettings+and+buildSettings) for the difference between projectSettings and buildingSettings. We don't need to copy everything.

Comment (AlanQuatermain) [reply plugin/src/main/scala/NGPlugin.scala]: Done.

File: helpers/src/main/java/com/linkedin/plugin/NGTestsBase.java (hunk 2 above)
Comment (AlanQuatermain) [reply helpers/src/main/java/com/linkedin/plugin/NGTestsBase.java]: Leaving deprecated for 2.5 release; removing in 2.6.

File: build.sbt (hunk 1 above)
Comment (AlanQuatermain) [reply build.sbt]: Done :)

File: helpers/src/main/java/com/linkedin/plugin/NGTestsBase.java (hunk 2 above)
Comment (bbarkley) [reply helpers/src/main/java/com/linkedin/plugin/NGTestsBase.java]: As @FranklinYinanDing points out support for Plugins is totally removed in 2.5. I think we should remove all references here too, including WithPlugins annotation, FakeApplicationFactoryArgs/Impl, tests, etc

Comment (AlanQuatermain) [reply helpers/src/main/java/com/linkedin/plugin/NGTestsBase.java]: Okay, that's been cleaned out now. The only remaining vestige is a parameter on a deprecated constructor for the args builder, so folks not using `WithPlugins` don't get bitten by bincompat issues.

File: plugin/src/main/scala/NGPlugin.scala [hunk 4]
```diff
@@ -20,17 +20,30 @@ import de.johoop.testngplugin.TestNGPlugin._
 
 object NGPlugin extends Plugin {
 
+  @deprecated("Use ngProjectSettings or ngBuildSettings instead", "2.5.0")
```
Comment (FranklinYinanDing) [inline plugin/src/main/scala/NGPlugin.scala]: "Use ngProjectSettings instead" ;)

File: project/plugins.sbt [hunk 5]
```diff
@@ -1 +1,3 @@
 addSbtPlugin("com.eed3si9n" % "sbt-doge" % "0.1.5")
+
+addSbtPlugin("de.johoop" % "sbt-testng-plugin" % "3.1.1")
```
Comment (FranklinYinanDing) [inline project/plugins.sbt]: Why is this needed?

File: sample/build.sbt [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import Defaults._
+import de.johoop.testngplugin.TestNGPlugin
+
+lazy val root = (project in file("."))
+  .settings(
+    name := "Sample",
+    version := "2.5.0-SNAPSHOT",
+    scalaVersion := "2.11.11",
+    crossScalaVersions := Seq("2.10.6", "2.11.11", "2.12.4"),
```
Comment (FranklinYinanDing) [inline sample/build.sbt]: I don't think "2.10.6" would work.

File: sample/build.sbt [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    libraryDependencies ++= Seq(
+      "com.linkedin.play-testng-plugin" %% "play-testng-helpers" % "2.5.0"
+    )
+  )
+  .enablePlugins(play.sbt.PlayJava)
+  .enablePlugins(TestNGPlugin)
```
Comment (FranklinYinanDing) [inline sample/build.sbt]: Why is this needed?

File: sample/project/plugins.sbt [hunk 8]
```diff
@@ -5,6 +5,8 @@ logLevel := Level.Warn
 resolvers += "Typesafe Releases Repository" at "http://repo.typesafe.com/typesafe/releases/"
 
 // Use the Play sbt plugin for Play projects
-addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.4.0")
+addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.5.0")
 
-addSbtPlugin("com.linkedin" % "play-plugins-testng" % "2.4.0")
+addSbtPlugin("de.johoop" % "sbt-testng-plugin" % "3.1.1")
```
Comment (FranklinYinanDing) [inline sample/project/plugins.sbt]: Why is this needed?

File: project/plugins.sbt (hunk 5 above)
Comment (AlanQuatermain) [reply project/plugins.sbt]: Without this, the TestNG-based tests weren't running in sbt. For Scala 2.11, I had to update from 3.0.2 to 3.1.1 to get it to build.

File: sample/build.sbt (hunk 7 above)
Comment (AlanQuatermain) [reply sample/build.sbt]: Without this, the TestNG-based tests weren't running in sbt. For Scala 2.11, I had to update from 3.0.2 to 3.1.1 to get it to build.

Note that I ported the whole thing from the old `Build.scala` to `build.sbt`.

Comment (AlanQuatermain) [reply sample/build.sbt]: Just verified: without enabling the TestNGPlugin, sbt doesn't see or execute the TestNG tests.

File: sample/project/plugins.sbt (hunk 8 above)
Comment (AlanQuatermain) [reply sample/project/plugins.sbt]: Okay, turns out I don't need to import the TestNG plugin directly.

File: project/plugins.sbt (hunk 5 above)
Comment (FranklinYinanDing) [reply project/plugins.sbt]: I just tried to remove it and it works. This `addSbtPlugin` is adding dependency for the library projects' `build.sbt`. Since the library projects (NGPlugin and NGHelper) do not have any TestNG tests, they do not need this in their project settings.

File: sample/build.sbt (hunk 6 above)
Comment (FranklinYinanDing) [reply sample/build.sbt]: Since Play 2.5 does not have 2.10, the sample project with 2.10 will fail.
